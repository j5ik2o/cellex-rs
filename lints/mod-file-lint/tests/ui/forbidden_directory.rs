#![feature(register_tool)]
#![register_tool(mod_file)]
#![warn(mod_file::no_mod_rs)]

#[path = "auxiliary/legacy/mod.rs"]
mod legacy;

fn main() {
  let _ = legacy::value();
}
