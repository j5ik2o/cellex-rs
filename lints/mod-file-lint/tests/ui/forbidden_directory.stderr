warning: mod.rs は Cellex のコーディング規約で使用禁止です
  --> $DIR/auxiliary/legacy/mod.rs:1:1
   |
LL | pub fn value() -> u32 {
   | ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: 対象ファイル: $DIR/auxiliary/legacy/mod.rs
   = help: ファイルを `<module>.rs` にリネームし、必要なら同ディレクトリに `<module>/` を作成してサブモジュールを移動してください
   = note: AI向けアドバイス: 1. `git mv foo/mod.rs foo.rs` 2. 親モジュールの `mod foo;` はそのまま利用 3. サブモジュールは foo.rs 内で `mod sub;` として再宣言しましょう。
   = note: `#[warn(no_mod_rs)]` on by default

warning: 1 warning emitted

